File: Java/Data Structures & Algorithms Module - Java/Revision/ArrayLists/ArrayListApp.java
/*
ArrayListApp.java
Billy Cussen
04/01/2020
*/

import java.util.List;
import java.util.ArrayList;

public class ArrayListApp{

    public static void main(String [] args){

        Person one = new Person("Billy","IT");
        Person two = new Student("William","Software");
        List<Person> myList = new ArrayList<Person>();
        myList.add(one);
        myList.add(two);

        for(Person e: myList){
            if(e instanceof Student){
                Student a = (Student) e;
                System.out.println(a.toString());
                System.out.println(a.isStudent());
            }
        }

    }

}

class Person{

    private String name;
    private String profession;

    Person(String name, String profession){
        this.name = name;
        this.profession = profession;
    }

    @Override
    public String toString(){
        return "Name: "+this.name+", Profession: "+this.profession;
    }

}

class Student extends Person{

    Student(String name, String profession){
        super(name, profession);
    }   

    public boolean isStudent(){
        return true;
    }

}

File: Python/Data Structures & Algorithms Module - Python/Revision/ArrayLists/ArrayListApp.py
"""
ArrayListApp.py
Billy Cussen
04/01/2020
"""

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def __str__(self):
        return "Name: "+self.name+", Age: "+str(self.age)

class Student(Person):
    def __init___(self, name, age):
        super().__init__(name, age)

    def isStudent(self):
        return True

one = Person("Billy", 31)
two = Student("William", 32)

myList = []
myList.append(one)
myList.append(two)

for e in myList:
    if isinstance(e,Student):
        print(str(e))
        print(str(e.isStudent()))
